src/controllers/api/thoughts.js

```diff
@@ -0,0 +1,89 @@
+const { Thought } = require("../../models");
+
+const getThoughts = async (req, res) => {
+  try {
+    const thoughts = await Thought.find({});
+    return res.json({ success: true, data: thoughts });
+  } catch (error) {
+    console.log(`[ERROR]: Failed to get thoughts | ${error.message}`);
+    return res
+      .status(500)
+      .json({ success: false, error: "Failed to get thoughts" });
+  }
+};
+
+const getThoughtById = async (req, res) => {
+  try {
+    const { thoughtId } = req.params;
+
+    const thought = await Thought.findById(thoughtId);
+    return res.json({ success: true, data: thought });
+  } catch (error) {
+    console.log(`[ERROR]: Failed to get thought | ${error.message}`);
+    return res
+      .status(500)
+      .json({ success: false, error: "Failed to get thought" });
+  }
+};
+
+const createThought = async (req, res) => {
+  try {
+    const { thoughtText, username } = req.body;
+
+    if (thoughtText && username) {
+      const newThought = await Thought.create({ thoughtText, username });
+      return res.json({ success: true, data: newThought });
+    }
+
+    return res.status(400).json({
+      success: false,
+      error: "Please provide the thoughtText and username",
+    });
+  } catch (error) {
+    console.log(`[ERROR]: Failed to create newThought | ${error.message}`);
+    return res
+      .status(500)
+      .json({ success: false, error: "Failed to create newThought" });
+  }
+};
+
+const updateThoughtById = async (req, res) => {
+  try {
+    const { thoughtId } = req.params;
+    const { thoughtText, username } = req.body;
```

Comment:
What if these two keys don't exist in the `req.body`. Make sure you have error handling (usually with `if` statements), to make sure the input matches what you are expecting.